src/UI/Components/UserCredentialsForm/index.tsx

```diff
@@ -0,0 +1,148 @@
+   * @param {string} value - The current value of the input field.
+   */
+  const handleUsernameChange = (
+    _event: React.FormEvent<HTMLInputElement>,
+    value: string,
+  ) => {
+    setUsername(value);
+  };
+
+  /**
+   * Handle the change of the password input field.
+   * @param {React.FormEvent<HTMLInputElement>} _event The event object.
+   * @param {string} value The current value of the input field.
+   */
+  const handlePasswordChange = (
+    _event: React.FormEvent<HTMLInputElement>,
+    value: string,
+  ) => {
+    setPassword(value);
+  };
+
+  return (
+    <Form className="loginForm">
+      {isError && error && (
+        <FormHelperText>
+          <HelperText>
+            <HelperTextItem
+              variant={isError ? "error" : "default"}
+              icon={<ExclamationCircleIcon />}
+              aria-label="error-message"
+            >
+              {error.message}
+            </HelperTextItem>
+          </HelperText>
+        </FormHelperText>
+      )}
+      <FormGroup label="Username" isRequired fieldId="pf-login-username-id">
+        <TextInput
+          id="pf-login-username-id"
+          isRequired
+          validated={ValidatedOptions.default}
+          type="text"
+          name="pf-login-username-id"
+          aria-label="input-username"
+          value={username}
+          onChange={handleUsernameChange}
+        />
+      </FormGroup>
+      <FormGroup label={"Password"} isRequired fieldId="pf-login-password-id">
+        {
+          <InputGroup>
+            <InputGroupItem isFill>
+              <TextInput
+                isRequired
+                type={isPasswordHidden ? "password" : "text"}
+                id="pf-login-password-id"
+                name="pf-login-password-id"
+                aria-label="input-password"
+                validated={ValidatedOptions.default}
+                value={password}
+                onChange={handlePasswordChange}
+              />
+            </InputGroupItem>
+            <InputGroupItem>
+              <Button
+                variant="control"
+                onClick={() => setIsPasswordHidden(!isPasswordHidden)}
+                aria-label={
+                  isPasswordHidden ? "show-password" : "hide-password"
+                }
+              >
+                {isPasswordHidden ? <EyeIcon /> : <EyeSlashIcon />}
+              </Button>
+            </InputGroupItem>
+          </InputGroup>
+        }
+      </FormGroup>
+      <ActionGroup>
+        <Button
+          aria-label={submitButtonLabel}
+          variant="primary"
+          type="submit"
+          onClick={(event) => {
+            event.preventDefault();
+            onSubmit(username, password);
+          }}
+          isBlock
+          isDisabled={isPending}
+        >
+          {isPending ? <Spinner size="md" /> : <Text>{submitButtonText}</Text>}
+        </Button>
+      </ActionGroup>
+    </Form>
+  );
+};
+
+export default UserCredentialsForm;
```

Comment:
I personally don't have a real problem with mixing default exports and named exports, but maybe we should keep it consistent?

Reply:
Fixed